src/providers/authProvider/permissions.ts

```diff
@@ -0,0 +1,68 @@
+type UserRoles = UserRole | 'user'
```

Comment:
This is an impressive module @salony75way - you've clearly done this before 🤣 

I wouldn't have had a clue where to start ;-)

Reply:
@IanMayo  Yes i did this once, But in react admin we have Role-Based Access Control (RBAC) library but its only in  enterprise edition so we have to create it custom.